gui/CMakeLists.txt

```diff
@@ -210,6 +209,7 @@ target_link_libraries(TeleSculptor
   kwiver::kwiver_adapter
   kwiver::kwiver_algo_core
   kwiver::kwiver_algo_mvg
+  kwiver_algo_qt_widgets
```

Comment:
Why does this not have `kwiver::`?

Reply:
I can try this again, but I believe it was causing linking errors with the `kwiver::`.  We had the same problem with `kwiver_algo_vtk` below.  When using the `kwiver::` prefix it would not be able locate some Qt or qtExtensions libraries that `kwiver_algo_qt_widgets` depends on.  Maybe this means we are exporting things incorrectly from KWIVER?  I'm not really sure how all this stuff works.

Reply:
The error I get with `kwiver::kwiver_algo_qt_widgets` is
```
 fatal error LNK1104: cannot open file 'Qt5::Widgets.lib'
```
Do you have any idea what we are doing wrong?

Reply:
...and *doesn't* happen without the namespace prefix? That's... concerning. I'd thought that the targets were explicitly aliases of each other, but it looks like *only* the prefixed names are actually targets, and the non-prefixed version is being translated into the name of an unknown library. If that's the case, that's bad, because we most assuredly want the imported targets to work.

I wonder if CMake is somehow not seeing the `Qt5::Widgets` target? But it appears later via `${QT_LIBRARIES}`, so I don't understand what could be going wrong.

I would look and see if there seems to be anything obviously wrong in the KWIVER imported targets, and if not, maybe ask someone with more CMake knowledge. Also, as a sanity check, can you link to `kwiver::kwiver_algo_qt`?

Reply:
>  Maybe this means we are exporting things incorrectly from KWIVER?

That's certainly possible. At least on Linux, it *looks* okay to me, but maybe it's different on Windows somehow. (I don't *think* the problem would be in qtExtensions, as that seems to be working for other projects, but I can't entirely discount the possibility.)

If you can find the `INTERFACE_LINK_LIBRARIES` for `kwiver::kwiver_algo_qt_widgets`, what do they look like?

Reply:
You are correct that it works properly without the namespace prefix.

If I link to `kwiver::kwiver_algo_qt` I get a similar error but for `Qt5::Gui.lib` instead.  I expect it's all Qt5 libraries that are not found but it stops after the first error.

We also saw analogous errors with the vtk arrow and VTK targets.  I believe @danlipsa first encountered this issue, and I believe he found it on Linux.  I'm hitting this issue on Windows.

in `kwiver-config-targets.cmake` I have
```
set_target_properties(kwiver::kwiver_algo_qt_widgets PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "C:/Users/matt.leotta/Software/kwiver/src;C:/Build/kwiver"
  INTERFACE_LINK_LIBRARIES "Qt5::Widgets;Qt5::Xml;qtExtensions;kwiver::vital"
)
```

Reply:
How is TeleSculptor supposed to get target definitions for targets that are imported into KWIVER and not defined by KWIVER?  Is KWIVER supposed to export any targets that it imports and links to publicly?

This is beyond my level of CMake understanding.

Reply:
> How is TeleSculptor supposed to get target definitions for targets that are imported into KWIVER and not defined by KWIVER? Is KWIVER supposed to export any targets that it imports and links to publicly?

That's a good question; how to handle transitive dependencies in CMake is currently not well defined, and often poorly implemented.

However, I am confused because AFAICT those targets _should_ already exist from TeleSculptor itself having issued a `find_package(Qt5 ...)`. And, as noted, those *exact same names* occur later — at least AFAICT; maybe this would be good to verify on your end? — via `${QT_LIBRARIES}`. We probably want to ask a CMake developer if they have any ideas why this wouldn't be working.

Reply:
BTW, my `INTERFACE_LINK_LIBRARIES` for `kwiver::kwiver_algo_qt` looks like `"kwiver::kwiver_algo_core;Qt5::Gui;..."`; I would expect yours is the same. That would match with your guess (with which I concur) that the error is from the first unrecognized target, but what's *odd* is that this implies that `kwiver::kwiver_algo_core` was resolved.

Reply:
Is `find_package(kwiver)` doing `find_package(Qt5 COMPONENTS …)` (or `find_dependency`) somewhere? Though the lack of an error on the target with `::` in it is odd. I've also heard of it before, but I'm having trouble remembering exactly where that was…

Reply:
My guess is the non-prefixed name is decaying into the same "gee, I hope I can just stick `-l` in front of that" that's causing problems when it happens to the `Qt5::` targets, and by luck, is working.

AFAIK, imported KWIVER does not have any form of `find_package(Qt5)`.

Reply:
The fix for this issues is on another PR, #407, I'm not going to address it on this branch.